drake/systems/plants/RigidBodySystem.cpp

```diff
@@ -134,9 +134,11 @@ RigidBodySystem::StateVector<double> RigidBodySystem::dynamics(
           tangent1 /= sqrt(this_normal(1) * this_normal(1) +
                            this_normal(0) * this_normal(0));
         }
-        Vector3d tangent2 = tangent1.cross(this_normal);
+        Vector3d tangent2 = this_normal.cross(tangent1);
         Matrix3d R;  // rotation into normal coordinates
-        R << tangent1, tangent2, this_normal;
+	R.row(0) = tangent1;
+	R.row(1) = tangent2;
+	R.row(2) = this_normal;
         auto J = R * (JA - JB);          // J = [ D1; D2; n ]
```

Comment:
In the absence of bulletproof notation, this kind of code needs careful commenting. I'd start by making sure every frame has a name. In this case, we're building a contact frame C, with Cx=tangent1, Cy=tangent2, Cz=this_normal. We have Jacobians JA and JB that map from joint space velocity to velocities in world (Cartesian) space W (not sure if that's what it's called in Drake). We're going to need a rotation matrix that maps from W to C. In Kane's notation when typeset, that would look like
```
 C   C W    W
v  =  R  * v
```
meaning that the vector on the right is expressed in W and re-expressed in C which can be seen by inspection of the superscripts, in which adjacent ones must always match. In code this can be flattened using "monogram" notation in which the important thing comes first and then the frame follow in order: `v_C = R_CW * v_W`, for example. Now a rotation matrix `R_AB` is constructed by expressing B's axes in A, and then arranging them in columns. In the code above the axes of C are expressed in W, so we have R_WC=[Cx Cy Cz]. By filling this in row-wise as you did above you constructed the transpose R_CW=R_WC' (Matlab notation).

My point is that the code is doing a lot of simple transformations and if each one is clearly documented (ideally with a good notation) then the whole thing has to work. Without that it usually comes down to trial and error!